src/osyris/core/vector.py

```diff
@@ -209,7 +210,9 @@ def __invert__(self):
         return np.logical_not(self)
 
     def to(self, unit):
-        return self.__class__(**{c: xyz.to(unit) for c, xyz in self._xyz.items()})
+        return self.__class__(**{c: xyz.to(unit)
+                                 for c, xyz in self._xyz.items()},
+                              name=self.name)
```

Comment:
Any reason why you need the name here? It's not added in the `.to` method for the `Array`, so why have it here?

Reply:
This might not be just pure aesthetics. Thinking about it, what happens if you fx comvert your positions to `pc` instead of `cm`. Then the name is empty, and the special case handling for the spherical components for any array that has the name `"position"` goes wrong?

Maybe this suggests we need a better mechanism for identifying vectors that are position vectors. Maybe using the name to decide is too brittle, and we need a dedicated member variable in the class?
But then how would that propagate in operations?

Reply:
The issue with assigning a dedicated member variable of the class to specify this is that nothing is really stopping the user from replacing it in the dataset (e.g doing `data["amr"]["position"] = data["amr"]["position"] - data["sink"]["position"][0]`. Unless this member is passed on during operations (probably a headache to implement), we will have `AttributeError`s when trying to check what kind of vector we're working on.

Reply:
> Unless this member is passed on during operations (probably a headache to implement)

I know it sounds like a headache. But think of the following:
```Py
data['amr']['pos_au'] = data['amr']['position'].to('au')
r = data['amr']['pos_au'].r
```
`r` here would be completely wrong, just because the name is `pos_au` and not `position`?
Maybe it's an indication that the approach with accessors that do something different based on the name is bad?

Reply:
I'm trying to see if we can fix this with a child class of Vector for positions...

In operations, can you think of any operation you could apply to a Vector (apart from the `.to` conversions) that would result in a new position vector?
It seems if you multiply a position with anything else it will no longer be a position vector?
Would you add a position to anything?
Maybe it's not such a headache? Maybe just having a member variable is not too bad?

Reply:
Well in your example, `r` would have the correct value, but I see your point; `.phi` and `.theta` would return wrong results.

Maybe we could simply check the dimensions of the vector? If it's a length then we can return radius, colatitude and azimuth whereas if it's any other dimension we return its spherical component?